ldb_reader.go

```diff
@@ -244,14 +313,30 @@ func (reader *LDBReader) Close() error {
 	reader.mu.Lock()
 	defer reader.mu.Unlock()
 
+	if reader.cancelWatcher != nil {
+		reader.cancelWatcher()
+	}
+
+	return reader.closeDB()
+}
+
+// closeDB closes all reader-owned resources associated with the current DB.
+// It should only be called when the caller is holding the reader.mu mutex.
+func (reader *LDBReader) closeDB() error {
 	for _, stmt := range reader.getRowByKeyStmtCache {
 		stmt.Close()
```

Comment:
I know this was how it was to start with but we might want to return an error if non-nil when closing these statements.